packages/adminrouter/extra/src/common/auth/open.lua

```diff
@@ -0,0 +1,181 @@
+local authcommon = require "common.auth.common"
+local jwt = require "resty.jwt"
+local util = require "common.util"
+
+local SECRET_KEY = nil
+
+local key_file_path = os.getenv("SECRET_KEY_FILE_PATH")
+if key_file_path == nil then
+    ngx.log(ngx.WARN, "SECRET_KEY_FILE_PATH not set.")
+else
+    ngx.log(ngx.NOTICE, "Reading secret key from `" .. key_file_path .. "`.")
+    SECRET_KEY = util.get_stripped_first_line_from_file(key_file_path)
+    if (SECRET_KEY == nil or SECRET_KEY == '') then
+        -- Normalize to nil, for simplified subsequent per-request check.
+        SECRET_KEY = nil
+        ngx.log(ngx.WARN, "Secret key not set or empty string.")
+    end
+    jwt:set_alg_whitelist({HS256=1})
+end
+
+local function validate_jwt_or_exit()
+    uid, err = authcommon.validate_jwt(SECRET_KEY)
+    if err ~= nil then
+        if err == 401 then
+            return authcommon.exit_401("oauthjwt")
+        end
+
+        -- Other error statuses go here...
+
+        -- Catch-all, normally not reached:
+        ngx.log(ngx.ERR, "Unexpected result from validate_jwt()")
+        ngx.status = ngx.HTTP_INTERNAL_SERVER_ERROR
+        return ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
+    end
+    return uid
+end
+
+
+local function do_authn_and_authz_or_exit()
+    local uid = validate_jwt_or_exit()
+
+    -- Authz using authn :)
```

Comment:
Short story: In Open user is authorised if and only if it's identity was transferred into IAM. So in a way - if you manage to authn, then you are also authzed.

Long story: https://mesosphere.slack.com/archives/G0W7PAKSS/p1490372266557793